rust: rustdoc reproducibility issue fix

rust: rustdoc reproducibility issue fix

The 'codegen-units' option split the crate into multiple compilation units for parallel compilation. Currently, this split is causing the rustdoc to generate differnt binary between the builds.
To fix this the codegen-units & the lto options are disabled.

More info about options:
https://doc.rust-lang.org/cargo/reference/profiles.html#codegen-units
https://doc.rust-lang.org/rustc/codegen-options/index.html#lto

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Sundeep KOKKONDA <sundeep.kokkonda@windriver.com>
---
--- a/.cargo/config.toml
+++ b/.cargo/config.toml
@@ -3,3 +3,7 @@

 [source.vendored-sources]
 directory = "vendor"
+
+[profile.release]
+codegen-units = 1
+

